pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go

```diff
@@ -342,11 +340,9 @@ func (e *AzureMonitorDatasource) executeQuery(ctx context.Context, logger log.Lo
 		return dataResponse
 	}
 
-	if _, ok := subscriptions[query.Subscription]; !ok {
-		retrieveSubscriptionDetails(e, cli, ctx, logger, tracer, query.Subscription, dsInfo.Routes["Azure Monitor"].URL, dsInfo.DatasourceID, dsInfo.OrgID)
-	}
+	subscription := e.retrieveSubscriptionDetails(cli, ctx, logger, tracer, query.Subscription, dsInfo.Routes["Azure Monitor"].URL, dsInfo.DatasourceID, dsInfo.OrgID)
```

Comment:
My thought here: do you happen to know how many more API calls this incurs?

Reply:
So if the subscription name was in the map already it will be one more additional call to azure.

Reply:
Sounds good to me :ship: